### PR TITLE
Celery: tracing integration

### PR DESCRIPTION
**Pull request by SpennyNDaJets**
## Summary
- Requires Python version 3.8 or newer.
- Support tracing auto-instrumention for the celery library.
- Optimizations for requests auto-instrumentation tracing.
- Removes `tracing_origins` configuration for requests library and passes by default.

## How did you test this change?
1. Start the fastapi e2e app (consumer and worker)
2. Hit the endpoint to start a celery job: `curl -H "x-highlight-request: cs5/cr5" http://localhost:8000/celery`
3. Confirm the tracing works on the Highlight UI
- [ ] Span added for publisher
- [ ] Span added for consumer
- [ ] Both spans are linked to same trace and session
4) Start the flask app
5) Hit the endpoint to make an external request: `curl -H "x-highlight-request: r39/r39" http://localhost:8085/external`
6) Confirm the tracing works on the Highlight UI
- [ ] Span created for request
- [ ] Span are linked to correct trace and session

https://www.loom.com/share/ae9fefe78caf4c67967a3b64d5a969a0?sid=4323f805-3dbf-49c2-8194-74726087a46d

## Are there any deployment considerations?
N/A

## Does this work require review from our design team?
N/A

### CODE REVIEW

File: sdk/highlight-py/e2e/highlight_flask/app.py [hunk 1]
```diff
@@ -12,10 +12,7 @@
 app = Flask(__name__)
 H = highlight_io.H(
     "1",
-    integrations=[
-        FlaskIntegration(),
-        RequestsIntegration(tracing_origins=["localhost", "highlight.io"]),
-    ],
+    integrations=[FlaskIntegration()],
```
Comment (Vadman97) [inline sdk/highlight-py/e2e/highlight_flask/app.py:15]: don't we still need to have `RequestsIntegration()` here to have that instrumentation enabled?

**Review by Vadman97 — APPROVED** [review]
:ship:

File: sdk/highlight-py/pyproject.toml [hunk 2]
```diff
@@ -50,8 +52,10 @@ pytest-cov = ">=4"
 pytest-asyncio = ">=0"
 loguru = ">=0"
 stripe = ">=6"
+redis = "^5.0.1"
```
Comment (Vadman97) [inline sdk/highlight-py/pyproject.toml:55]: could we add a note to the e2e readme on how to run the fastapi / celery example and how we need to run redis for this to work?

File: sdk/highlight-py/e2e/highlight_flask/app.py (hunk 1 above)
Comment (SpennyNDaJets) [reply sdk/highlight-py/e2e/highlight_flask/app.py:15]: Nope! This is instrumented by default if the appropriate libraries are installed. We just had this here before to define the `tracing_origins`, which are no longer used

**Pull request merged**